rest-api/data_gen/fake_participant_generator.py

```diff
@@ -574,12 +602,69 @@ def _choose_name(self, answer_map):
     answer_map[EMAIL_QUESTION_CODE] = _string_answer(email)
 
   def _choose_date_of_birth(self, answer_map):
-    if random.random() <= _QUESTION_NOT_ANSWERED:
-      return
     delta = datetime.timedelta(days=random.randint(0, self._max_days_for_birth_date))
     date_of_birth = (self._min_birth_date + delta).date()
     answer_map[DATE_OF_BIRTH_QUESTION_CODE] = [{"valueDate": date_of_birth.isoformat()}]
 
+  def _choose_answer_for_spec(self, answer_spec, answer_count):
+    # We assume that each question only has one type of answer
+    if int(answer_spec['code_answer_count']) > 0:
+      codes = answer_spec['code_answers'].split(',')
+      if answer_count == 1:
+        return [_code_answer(random.choice(codes))]
+      else:
+        return [_code_answer(code) for code in random.sample(codes, answer_count)]
+    if int(answer_spec['decimal_answer_count']) > 0:
+      return [{'valueDecimal': round(random.uniform(float(answer_spec['min_decimal_answer']),
+                                                    float(answer_spec['max_decimal_answer'])), 1)}
+              for _ in xrange(answer_count)]
+    if int(answer_spec['integer_answer_count']) > 0:
+      return [{'valueInteger': random.randint(int(answer_spec['min_integer_answer']),
+                                              int(answer_spec['max_integer_answer']))}
+              for _ in xrange(answer_count)]
+    if int(answer_spec['date_answer_count']) > 0:
+      min_date = parse(answer_spec['min_date_answer'])
+      max_date = parse(answer_spec['max_date_answer'])
+      days_diff = (max_date - min_date).days
+      return [{'valueDate': (min_date +
+                             datetime.timedelta(days=random.randint(0, days_diff))).isoformat()}
+              for _ in xrange(answer_count)]
+    if int(answer_spec['datetime_answer_count']) > 0:
+      min_date = parse(answer_spec['min_datetime_answer'])
+      max_date = parse(answer_spec['max_datetime_answer'])
+      seconds_diff = (max_date - min_date).total_seconds()
+      return [{'valueDateTime': (min_date +
+                                 datetime.timedelta(seconds=random.randint(0, seconds_diff)))
+                                 .isoformat()}
+              for _ in xrange(answer_count)]
+    if int(answer_spec['boolean_answer_count']) > 0:
+      return [{'valueBoolean': random.random() < 0.5}]
+    if int(answer_spec['string_answer_count']) > 0:
+      return [{'valueString': ''.join([random.choice(string.lowercase) for _ in xrange(20)])}
+              for _ in xrange(answer_count)]
+    if int(answer_spec['uri_answer_count']) > 0:
```

Comment:
Can we make this example.com or something else that obviously won't point to data?

Reply:
Done.